src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -180,6 +269,21 @@ private void IssueCommand(byte[] data)
             currentSlave.Write(data);
         }
 
+        private bool SocketConnected(Socket s)
+        {
+            bool part1 = s.Poll(1000, SelectMode.SelectRead);
+            bool part2 = (s.Available == 0);
+            if (part1 && part2)
+                return false;
+            else
+                return true;
+        }
+        
+        private int GetLength(byte b1, byte b2){
+            int combined = b1 << 8 | b2;
+            return combined;
+        }
```

Comment:
Can you make this a nested function in `StartSocketServer`? Not sure if C# allows this, maybe it allows lambdas.

Reply:
A nested function should be okay

Reply:
Seems that for some reason, C# (at least in this version) doesn't support nested function